core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -3185,16 +3197,106 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
       "Spark can only do this while using the new shuffle block fetching protocol"))
   }
 
+  test("SPARK-51272: retry all the partitions of result stage, if the first result task" +
+    " has failed and ShuffleMap stage is inDeterminate") {
+    var resubmitFailedStageTriggered = false
+    val monitor = new Object()
```

Comment:
You can remove these two lines. No monitor is needed as the event order triggers the error and enough to
post a `CompletionEvent` right before the `ResubmitFailedStages`. Those posts happens on the same thread, see how `DagSchedulerInterceptor` is used in the `post` method.

Reply:
The reason is mentioned above.. Lets discuss if you think otherwise..